src/Mailgun/Api/AbstractApi.php

```diff
@@ -39,6 +44,25 @@ public function __construct(HttpClient $httpClient, RequestFactory $requestFacto
     }
 
     /**
+     * Attempts to safely deserialize the response into the given class.
+     * If the HTTP return code != 200, deserializes into SimpleResponse::class
+     * to contain the error message and any other information provided.
+     *
+     * @param ResponseInterface $response
+     * @param string            $className
+     *
+     * @return $class|SimpleResponse
+     */
+    protected function safeDeserialize(ResponseInterface $response, $className)
```

Comment:
Most of HTTP Clients are offering a `->isSuccessful()` method and it checks if the status code is `>= 200 && < 300` so maybe - not sure yet - we could use this method (or this `if` statement) to provide the expected behaviour. What is your thought about this?